src/controller/postUrl.ts

```diff
@@ -0,0 +1,28 @@
+import express, { NextFunction } from "express";
+import { databaseModel } from "../model";
+import { logger } from "../services";
+import { generateCustomName, STATUS_CODES } from "../utils";
+import { handleError, sendResponse } from "../utils";
+
+export const postUrl = async (
+  req: express.Request,
+  res: express.Response,
+  next: NextFunction,
+) => {
+  try {
+    const { originalUrl, customName } = req.body;
+
+    const shortUrl = await databaseModel.create({
+      originalUrl,
+      shortUrl: customName || generateCustomName(),
```

Comment:
Since default is already specified in the model, I don't think this is needed. Verify though
```suggestion
      shortUrl: customName,
```